azurerm/internal/services/loganalytics/log_analytics_linked_storage_account_resource.go

```diff
@@ -0,0 +1,170 @@
+package loganalytics
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/operationalinsights/mgmt/2020-03-01-preview/operationalinsights"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/loganalytics/parse"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmLogAnalyticsLinkedStorageAccount() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmLogAnalyticsLinkedStorageAccountCreateUpdate,
+		Read:   resourceArmLogAnalyticsLinkedStorageAccountRead,
+		Update: resourceArmLogAnalyticsLinkedStorageAccountCreateUpdate,
+		Delete: resourceArmLogAnalyticsLinkedStorageAccountDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.LogAnalyticsLinkedStorageAccountID(id)
+			return err
+		}),
+
+		Schema: map[string]*schema.Schema{
+			"data_source_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					strings.ToLower(string(operationalinsights.CustomLogs)),
+					strings.ToLower(string(operationalinsights.AzureWatson)),
+					strings.ToLower(string(operationalinsights.Query)),
+					strings.ToLower(string(operationalinsights.Ingestion)),
+					strings.ToLower(string(operationalinsights.Alerts)),
+				}, false),
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"workspace_resource_id": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateFunc:     azure.ValidateResourceID,
+				DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
we should link to an issue describing [this](https://github.com/Azure/azure-rest-api-specs/issues/9633):
```suggestion
				https://github.com/Azure/azure-rest-api-specs/issues/9633
				DiffSuppressFunc: suppress.CaseDifference,
```